### PR TITLE
[Improvement-15713][api]DataSource And UdfFunc list query use Enum code value rather than ordinal

### LINKED ISSUE #15713: [Improvement][api]DataSource And UdfFunc list query shold use Enum code not ordinal
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

When saving DataSource entities to the database, the DbType value is determined by the field annotated with @EnumValue, where it takes the value of 'code'. Similarly, when saving UdfFunc entities, the 'type' field operates in a similar manner. However, during list queries in the API module, the values are obtained using the ordinal method of the enums.

During the process of extending DbType in DolphinScheduler's secondary development, users have observed that if the codes are not incremented properly, bugs may occur.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by songwenyong**
## Purpose of the pull request

fix #15713

## Brief change log

- DataSource list query use DbType enum code value rather than ordinal

- UdfFunc list query use UdfType code value rather than ordinal

## Verify this pull request

This pull request is code cleanup without any test coverage.

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by rickchengx — COMMENTED** [review]
@songwenyong  please link this PR to an issue

**Comment by songwenyong** [comment]
> @songwenyong please link this PR to an issue

#15713

**Comment by songwenyong** [comment]
> @songwenyong please link this PR to an issue

done

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ public Result<Object> queryDataSource(@Parameter(hidden = true) @RequestAttribut
     @ApiException(QUERY_DATASOURCE_ERROR)
     public Result<Object> queryDataSourceList(@Parameter(hidden = true) @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
                                               @RequestParam("type") DbType type) {
-        List<DataSource> datasourceList = dataSourceService.queryDataSourceList(loginUser, type.ordinal());
+        List<DataSource> datasourceList = dataSourceService.queryDataSourceList(loginUser, type.getCode());
```
Comment (rickchengx) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java:165]: overall LGTM, please update the related UT

**Review by fuchanghai — APPROVED** [review]
LGTM

**Pull request merged**